apps/io-func-wallet-solution-backend/src/infra/http/handlers/get-nonce.ts

```diff
@@ -0,0 +1,16 @@
+import * as H from "@pagopa/handler-kit";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { logErrorAndReturnResponse } from "../utils";
+import { insertNonce, generateNonce } from "@/nonce";
+
+export const GetNonceHandler = H.of(() =>
+  pipe(
+    generateNonce,
+    RTE.fromIOEither,
+    RTE.chainFirstW(insertNonce),
```

Comment:
What do you think about inserting a method to check if the generated nonce is already present in the DB, then generating another one if so? It's a very remote probability but it could still happen

Reply:
if it's an insert on ID cosmos should return 409 as status code on conflicting IDs

Reply:
Ok thanks, I was wondering if it would make sense to retry directly on the backend or leave it to the end user (IO app)

Reply:
Do we want to implement retry directly in the backend?

Reply:
Let's keep this as a discussion point. Do you think it makes sense?

Reply:
yes it could

Reply:
The Cosmos SDK features automatic retries. However, in any scenario where the client has the ability to retry, the responsibility should remain with the client.

Reply:
We thought about it more carefully. In fact, the nonce in the DB have a TTL in the order of minutes, therefore the probability that a random string is equal to one generated a few minutes ago is so low that we can ignore this. So I would say not to add the retry and still leave an error in case the insert is not successful.